src/python/pants/backend/shell/target_types.py

```diff
@@ -384,6 +384,29 @@ class ShellCommandWorkspaceInvalidationSourcesField(AdhocToolWorkspaceInvalidati
     pass
 
 
+class PathShimsMode(Enum):
+    PREPEND = "prepend"
+    APPEND = "append"
+    OFF = "off"
+
+
+class ShellCommandPathShimsModeField(StringField):
+    alias = "path_shims_mode"
```

Comment:
This is the PATH mode for shims, right, so to me `shims_path_mode` feels more accurate as name for this. wdyt?

Reply:
I like the idea from ordering perspective. Maybe `tools_path_mode` instead since `tools` is the field which is (partially) driving the `PATH` changes?

Reply:
good point.

Reply:
I ended up renaming it to `path_env_modify` to reflect that it controls how the `PATH` environment variable is modified.

Reply:
cc @kaos - How does `path_env_modify` sound to you?